src/Adapter/LegacyContext.php

```diff
@@ -43,6 +44,25 @@
  */
 class LegacyContext
 {
+    /** @var Currency */
+    private $employeeCurrency;
+
+    /** @var string */
+    private $mailThemesUri;
+
+    /** @var Tools */
+    private $tools;
+
+    /**
+     * This constructor must remain empty to avoid BC breaks. If you need to
+     * inject services please implement a setter and use the method injection
+     * instead.
+     */
+    public function __construct()
+    {
+        $this->tools = new Tools();
```

Comment:
Tools class have only static function.

Reply:
No, because this is actually the `PrestaShop\PrestaShop\Adapter\Tools` class (which I'm sure you didn't know about either ^^)
not saying this is pretty, but at least it prevents a dependency on legacy class